apps/website/src/pages/api/auth/[...nextauth].ts

```diff
@@ -27,9 +31,33 @@ type ProfileData = {
 };
 
 export const authOptions: NextAuthOptions = {
-  // Include user.id on session
   callbacks: {
     session: async function ({ session, user }) {
+      // Check the user's access token is valid, refresh if not
+      const token = await prisma.account.findFirst({
+        where: { userId: user.id, provider: "twitch" },
+        select: { id: true, access_token: true, refresh_token: true },
+      });
+      if (token?.access_token && token?.refresh_token) {
+        try {
+          await refreshAccessToken(
+            "twitch",
+            env.TWITCH_CLIENT_ID,
+            env.TWITCH_CLIENT_SECRET,
+            token.id,
+            token.access_token,
+            token.refresh_token,
+          );
```

Comment:
I am not sure we should run refreshAccessToken which will validate the token with twitch every time the session is accessed!?

But I am also not sure whats the best approach. Maybe we need to keep track of the last token update and check `validated_at + 1h < now` or something?

Reply:
Yeh, it might be a bit aggressive if folks are clicking in/out of the site continually. I wonder if we can just include a value in the session itself to track last validated? Not sure if extra data there persists or not.

Reply:
Turns out you can't store data in the session that persists, so put it in the DB.